QuickFIXn/Fields/Converters/DateTimeConverter.cs

```diff
@@ -9,260 +10,698 @@ namespace QuickFix.Fields.Converters;
     /// <summary>
-    /// Check if string is TimeOnly and, if yes, convert to DateTime.
+    /// Converts the specified span to a <see cref="DateOnly"/>.
+    /// The span must be in the format "yyyyMMdd".
     /// </summary>
-    /// <param name="str"></param>
-    /// <returns></returns>
-    /// <exception cref="FieldConvertError"/>
-    public static DateTime ParseToTimeOnly(string str)
+    /// <param name="str">A span containing the characters that represent a date to convert.</param>
+    /// <returns>
+    /// A value representing the date in <paramref name="str"/>.
+    /// </returns>
+    /// <exception cref="FieldConvertError">The conversion cannot be performed successfully.</exception>
+    public static DateOnly ParseToDateOnly(ReadOnlySpan<char> str)
     {
-        try {
-            DateTime d = (str.Length > TIME_ONLY_MAXLENGTH_WITHOUT_NANOSECONDS)
-                ? TimeOnlyFromNanoString(str)
-                : DateTime.ParseExact(str, TIME_ONLY_FORMATS, DATE_TIME_CULTURE_INFO, DATE_TIME_STYLES);
-
-            return new DateTime(1980, 1, 1) + d.TimeOfDay;
-        }
-        catch (Exception e)
+        if (!DateOnly.TryParseExact(str, "yyyyMMdd", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateOnly dateOnly))
         {
-            throw new FieldConvertError("Could not convert string (" + str + ") to TimeOnly: " + e.Message, e);
+            throw new FieldConvertError($"Could not convert string to DateOnly: {str}");
         }
+
+        return dateOnly;
     }
 
     /// <summary>
-    /// Gets the nanoseconds component of the date represented by this instance truncated to 100 nanosecond resolution
+    /// Converts the specified string to a <see cref="DateTime"/>.
+    /// The string must be in the format "yyyyMMdd".
+    /// </summary>
+    /// <param name="str">A <see cref="string"/> containing the characters that represent a date to convert.</param>
+    /// <returns>
+    /// A value representing the date in <paramref name="str"/> with its time component equal to 00:00:00 and
+    /// its <see cref="DateTime.Kind"/> equal to <see cref="DateTimeKind.Unspecified"/>.
+    /// </returns>
+    /// <remarks>
+    /// This method calls <see cref="ParseToDateOnly(ReadOnlySpan{char})"/> which returns a <see cref="DateOnly"/>.
+    /// Consider calling the latter for correctness.
+    /// </remarks>
+    /// <exception cref="FieldConvertError">The conversion cannot be performed successfully.</exception>
+    public static DateTime ParseToDateOnly(string str) => ParseToDateOnly((ReadOnlySpan<char>)str).ToDateTime(default);
+
+    /// <summary>
+    /// Gets the nanoseconds component of the date represented by this instance truncated to 100 nanosecond resolution.
     /// </summary>
     /// <param name="dt"></param>
     /// <returns>The nanoseconds component, expressed as a value between 0 and 99900.</returns>
-    private static int Nanosecond(this DateTime dt)
+    public static int Nanosecond(this DateTime dt)
```

Comment:
The engine doesn't use this.  Is there a compelling reason to retain it?

Reply:
No, not sure why I made it public

Reply:
Also, I see DateTime now has a Nanosecond property, so this is redundant anyway